components/tidb_query/src/batch/runner.rs

```diff
@@ -452,4 +461,140 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
     pub fn collect_storage_stats(&mut self, dest: &mut SS) {
         self.out_most_executor.collect_storage_stats(dest);
     }
+
+    // TODO: do not copy from the origin
+    pub fn handle_streaming_request(
+        &mut self,
+    ) -> Result<(Option<(StreamResponse, IntervalRange)>, bool)> {
+        let mut batch_size = BATCH_INITIAL_SIZE;
+        let mut warnings = self.config.new_eval_warnings();
+
+        // if the data read finished, is_drained means "finished"
+        let (mut record_cnt, mut is_drained) = (0, false);
+        let mut already_read_cnt = 0;
+        let mut chunk = Chunk::default();
+
+        // record count less than batch size and is not drained
+        info!(
+            "handle_streaming_request in Batch called, self.stream_batch_row_limit is {}",
+            self.stream_batch_row_limit
+        );
+        while record_cnt < 32 && already_read_cnt < 32 && !is_drained {
```

Comment:
Prefer `self.out_most_executor.next_batch(stream_batch_row_limit)` directly instead of this `while` loop.

Reply:
> Prefer `self.out_most_executor.next_batch(stream_batch_row_limit)` directly instead of this `while` loop.

1. Here we want the final result both no less than `stream_batch_row_limit` and not be a lot more than the real result.
2. `self.out_most_executor.next_batch(stream_batch_row_limit)` will fetch `stream_batch_row_limit` `physical_columns` , but the logical columns are tend to be less than `stream_batch_row_limit`.